cpp/include/ray/api/serializer.h

```diff
@@ -0,0 +1,77 @@
+
+#pragma once
+
+#include <msgpack.hpp>
+
+namespace ray {
+namespace api {
+
+class Serializer {
+ public:
+  template <typename T>
+  static void Serialize(msgpack::packer<msgpack::sbuffer> &packer, const T &val);
+
+  static void Serialize(msgpack::packer<msgpack::sbuffer> &packer);
+
+  template <typename Arg1Type, typename... OtherArgTypes>
+  static void Serialize(msgpack::packer<msgpack::sbuffer> &packer, const Arg1Type &arg1,
```

Comment:
what is the difference of these 2 methods? Can you add some comments?

Reply:
The first one is to zero arg and the second one is to at least one arg.